### PR TITLE
Support generic vault request params

### PR DESCRIPTION
**Pull request by djoksimo**
## What

- Add support for generic `BaseVaultRequestParams`
- [x] Fix broken unit tests

## Why

- Easier to support pos-specific params while reusing existing facilities

## Test Plan

- ✅ Updated unit tests to use the new generic params

## How

Can be merged as-is and other branches for POS refund + balance will build on top of these changes

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/network/data/BaseVaultRequestParams.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+) {
+    override fun equals(other: Any?): Boolean {
+        return other is BaseVaultRequestParams &&
+            other.cardNumberToken == cardNumberToken &&
+            other.encryptionKey == encryptionKey
+    }
+
+    override fun hashCode(): Int {
+        return cardNumberToken.hashCode() + encryptionKey.hashCode()
+    }
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/data/BaseVaultRequestParams.kt:15]: `data classes` do not support inheritance.

The overridden `equals` and `hashcode` are needed when `BaseVaultRequestParams` is used inside a `Map` (like in our unit tests, and possible other places in the future).

Open to learning about alternative workarounds.

File: forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt [hunk 2]
```diff
@@ -349,12 +348,27 @@ internal class VGSPinCollector(
                 .create()
         }
 
-        private fun buildRequestBody(cardToken: String): HashMap<String, String> {
-            val body = HashMap<String, String>()
-            body[ForageConstants.RequestBody.CARD_NUMBER_TOKEN] = cardToken
+        internal fun buildRequestBody(requestBodyParams: BaseVaultRequestParams): HashMap<String, Any> {
```
Comment (dleis612) [inline forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt:351]: Nice :)

File: forage-android/src/main/java/com/joinforage/forage/android/network/data/CapturePaymentRepository.kt [hunk 3]
```diff
@@ -59,22 +59,24 @@ internal class CapturePaymentRepository(
...
+                        PaymentMethod.ModelMapper.from(response.data)
+                    ),
+                    encryptionKey = encryptionKey
+                )
             )
             else -> response
         }
     }
 
     private suspend fun submitPaymentCapture(
```
Comment (dleis612) [inline forage-android/src/main/java/com/joinforage/forage/android/network/data/CapturePaymentRepository.kt:73]: I think this signature needs `<T : BaseVaultRequestParams>`

**Review by dleis612 — CHANGES_REQUESTED** [review]

**Review by dleis612 — APPROVED** [review]

**Review by devinmorgan — APPROVED** [review]
PR looks good! One thing I wanted to clarify:

From what I can tell, this PR only updates `VGSPinCollector` shuttle the the pos related data into the proxy request and this PR does not update the `BTPinCollector` to do the same. Is that correct?

**Pull request merged**